schedulezoomtalks.py

```diff
@@ -0,0 +1,115 @@
+import os
+import github
+import requests
+import secrets
+from researchseminarsdotorg import *
+from requests import get, post, put
+from io import StringIO
+from ruamel.yaml import YAML
+from common import *
+
+def schedule_zoom_talk(talk) -> Tuple[string, string]:
+
+    # Generate a password for the meeting. This is required since
+    # otherwise the meeting room will be forced. Zoom limits the
+    # password length to max 10 characters.
+    zoom_password = secrets.token_urlsafe(10)
+
+    # Form the talk registration body
+    request_body =
+    {
+      "topic": "Speakers\' corner talk by %s"%(talk.get("name")),
+      "type": 2, # Scheduled meeting
+      "start_time": talk["time"],
+      "timezone": "UTC",
+      "duration": 60,
+      "schedule_for": SPEAKERS_CORNER_USER_ID,
+      "password": zoom_password,
+
+      # Meeting settings
+      "settings": {
+        "host_video": True,
+        "participant_video": False,
+        "cn_meeting": False,  # Host the meeting in China?
+        "in_meeting": False,  # Host the meeting in India?
+        "join_before_host": False, # This will be switched to True shortly
+                                   # before the meeting starts by the VSF bot.
+                                   # It will also be switched back to False
+                                   # after the meetings ends.
+        "mute_upon_entry": True,
+        "watermark": False,  # Don't add a watermark when screensharing
+        "use_pmi": False, # Don't use Personal Meeting ID, but generate one
+        "approval_type": 0, # Automatically approve
+        "close_registration" : True, # Close registration after event date
+        "waiting_room" : False,    # No waiting room
+        "audio": "both",
+        "auto_recording": "cloud",
+        "enforce_login": False,
+        "alternative_hosts": "",
+        "registrants_email_notification": True,
+        "contact_email": "vsf@virtualscienceforum.org",
+      }
+    }
+
+    response = zoom_request(
+        requests.post,
+        f"{ZOOM_API}users/{user_id}/meetings",
+        params={"body":request_body}
+    )
+
+    return response.id, zoom_password
+
+def parse_talks(talks) -> int:
+    num_updated = 0
+    for talk in talks:
+        # If we are processing a speakers corner talk, and its meeting id has
+        # already been set, there's nothing left to do
+        if not (talk.get('zoom_meeting_id', "") == "") and
```

Comment:
`if "zoom_meeting_id" in talk or talk["event_type"] != "speakers_corner"`.

It's good to rely on data format being homogeneous, so to let's assume that zoom_meeting_id is optional but may not be blank, and event_type is required.